File: labs java/lab08/polymorphism/Polymorphism.cpp
#include <iostream>
using namespace std;

class Person{
    public:
    string greet(){
        return "I'm a person";
    }
};

class Student : Person{
    public:
    string greet(){
        return "I'm a Student";
    }
};

void process (Person p )


int main(){
    Person p1; //no polymorphism

    /* C++ polymorphism requires refrence or pointers */
    Person* p2 = new Student(); //polymorphism

    process(p1);
    process(p2);

    cout<<p1.greet()<<endl;
    cout<<p2.greet()<<endl;

}

File: labs java/lab08/polymorphism/Polymorphism.java
class Person {
    private String name;

    /*
     * public Person(String name){
     * this.name = name;
     * }
     */

    public String greet() {
        return "I'm a Person";
    }
}

class Student extends Person {
    public String greet() {
        return "I'm a Student";
    }

}

public class Polymorphism {
    public static void main(String[] args) {
        Person p1 = new Person(); // no polymorphism
        Person p2 = new Student();// polymorphism , upcasting
        process(p1);
        process(p2);
    }

    public static void process(Person p) {
        System.out.println("Processing " + p.greet());
    }
}